### PR TITLE
Adding a bound on EventHubSystemConsumer's initalizeEventHubsManagers

### PR DESCRIPTION
**Pull request by rmatharu**
Being a sync call, eventHubManagers can cause the container-start or the runLoop to block. 
Like so, 
"Samza StreamProcessor Container Thread-0" # 340 daemon prio=5 os_prio=0 tid=0x00007f5e38427800 nid=0x579e waiting on condition [0x00007f5e888cc000]
   java.lang.Thread.State: WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x00000005c1e04bf0> (a java.util.concurrent.CompletableFuture$Signaller)
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
	at java.util.concurrent.CompletableFuture$Signaller.block(CompletableFuture.java:1693)
	at java.util.concurrent.ForkJoinPool.managedBlock(ForkJoinPool.java:3323)
	at java.util.concurrent.CompletableFuture.waitingGet(CompletableFuture.java:1729)
	at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1895)
	at com.microsoft.azure.eventhubs.EventHubClient.lambda$createReceiverSync$7(EventHubClient.java:336)
	at com.microsoft.azure.eventhubs.EventHubClient$$Lambda$371/65833266.execute(Unknown Source)
	at com.microsoft.azure.eventhubs.impl.ExceptionUtil.sync(ExceptionUtil.java:191)
	at com.microsoft.azure.eventhubs.EventHubClient.createReceiverSync(EventHubClient.java:336)
	at org.apache.samza.system.eventhub.consumer.EventHubSystemConsumer.initializeEventHubsManagers(EventHubSystemConsumer.java:320)
	- locked <0x00000005c1e04cd0> (a com.linkedin.samza.eventhub.consumer.LiEventHubSystemConsumer)
	at org.apache.samza.system.eventhub.consumer.EventHubSystemConsumer.start(EventHubSystemConsumer.java:355)
	at org.apache.samza.system.SystemConsumers$$anonfun$start$5.apply(SystemConsumers.scala:168)
	at org.apache.samza.system.SystemConsumers$$anonfun$start$5.apply(SystemConsumers.scala:168)
	at scala.collection.Iterator$class.foreach(Iterator.scala:727)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1157)
	at scala.collection.MapLike$DefaultValuesIterable.foreach(MapLike.scala:206)
	at org.apache.samza.system.SystemConsumers.start(SystemConsumers.scala:168)
	at org.apache.samza.container.SamzaContainer.startConsumers(SamzaContainer.scala:999)
	at org.apache.samza.container.SamzaContainer.run(SamzaContainer.scala:779)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

This PR introduces a bound on it.

### CODE REVIEW

File: samza-azure/src/main/java/org/apache/samza/system/eventhub/consumer/EventHubSystemConsumer.java [hunk 1]
```diff
@@ -254,6 +256,13 @@ private EventHubClientManager createOrGetEventHubClientManagerForSSP(String stre
     return eventHubClientManager;
   }
 
+  private void boundedInitializeEventHubsManagers() {
```
Comment (dxichen) [inline samza-azure/src/main/java/org/apache/samza/system/eventhub/consumer/EventHubSystemConsumer.java]: Could you also add this logic to the producer? It should may encounter the same problem. https://github.com/apache/samza/blob/a702b3bb4a159804a2f7234710231b0dbf570b90/samza-azure/src/main/java/org/apache/samza/system/eventhub/producer/EventHubSystemProducer.java#L167

File: samza-azure/src/main/java/org/apache/samza/system/eventhub/consumer/EventHubSystemConsumer.java [hunk 2]
```diff
@@ -105,6 +106,7 @@
 
   // Overall timeout for EventHubClient exponential backoff policy
   private static final Duration DEFAULT_EVENTHUB_RECEIVER_TIMEOUT = Duration.ofMinutes(10);
+  private static final long DEFAULT_INITIALIZE_EVENTHUBSMANAGERS_TIMEOUT_MILLIS = Duration.ofSeconds(15).toMillis();
   private static final long DEFAULT_SHUTDOWN_TIMEOUT_MILLIS = Duration.ofSeconds(15).toMillis();
```
Comment (dxichen) [inline samza-azure/src/main/java/org/apache/samza/system/eventhub/consumer/EventHubSystemConsumer.java]: On second thought, this timeout is too low; the shutdown is async of all partitions. The `DEFAULT_EVENTHUB_RECEIVER_TIMEOUT` would be a more conservative value. @lhaiesp What do you think?

File: samza-azure/src/main/java/org/apache/samza/system/eventhub/consumer/EventHubSystemConsumer.java [hunk 3]
```diff
@@ -254,6 +256,13 @@ private EventHubClientManager createOrGetEventHubClientManagerForSSP(String stre
     return eventHubClientManager;
   }
 
+  private void boundedInitializeEventHubsManagers() {
+    if (!ShutdownUtil.boundedExecution(Collections.singletonList(() -> initializeEventHubsManagers()),
```
Comment (dxichen) [inline samza-azure/src/main/java/org/apache/samza/system/eventhub/consumer/EventHubSystemConsumer.java]: Would have to rename `ShutdownUtil` to use it here.

File: samza-azure/src/main/java/org/apache/samza/system/eventhub/consumer/EventHubSystemConsumer.java (hunk 1 above)
Comment (rmatharu) [reply samza-azure/src/main/java/org/apache/samza/system/eventhub/consumer/EventHubSystemConsumer.java]: done

File: samza-azure/src/main/java/org/apache/samza/system/eventhub/consumer/EventHubSystemConsumer.java (hunk 3 above)
Comment (rmatharu) [reply samza-azure/src/main/java/org/apache/samza/system/eventhub/consumer/EventHubSystemConsumer.java]: didn't need to, found a better way

File: samza-azure/src/main/java/org/apache/samza/system/eventhub/consumer/EventHubSystemConsumer.java (hunk 2 above)
Comment (rmatharu) [reply samza-azure/src/main/java/org/apache/samza/system/eventhub/consumer/EventHubSystemConsumer.java]: done

**Comment by lhaiesp** [comment]
Looks good

**Pull request closed without merging**